x/staking/keeper/keeper.go

```diff
@@ -226,6 +236,46 @@ func NewKeeper(
 		),
 		// key is: 113 (it's a direct prefix)
 		Params: collections.NewItem(sb, types.ParamsKey, "params", codec.CollValue[types.Params](cdc)),
+
+		// key format is: 101 | valAddr | uint64
+		ValidatorConsPubKeyRotationHistory: collections.NewMap(
+			sb, types.ValidatorConsPubKeyRotationHistoryKey,
+			"cons_pub_rotation_history",
+			collections.PairKeyCodec(collections.BytesKey, collections.Uint64Key),
+			codec.CollValue[types.ConsPubKeyRotationHistory](cdc),
+		),
+
+		// key format is: 102 | height
+		BlockConsPubKeyRotationHistory: collections.NewMap(
+			sb, types.BlockConsPubKeyRotationHistoryKey,
+			"cons_pubkey_history_by_block",
+			collections.Uint64Key,
+			codec.CollValue[types.ConsPubKeyRotationHistory](cdc),
+		),
+
+		// key format is: 103 | valAddr | time
+		ValidatorConsensusKeyRotationRecordIndexKey: collections.NewMap(
+			sb, types.ValidatorConsensusKeyRotationRecordQueueKey,
+			"cons_pub_rotation_index",
+			collections.PairKeyCodec(collections.BytesKey, sdk.TimeKey),
+			collections.BytesValue,
+		),
+
+		// key format is: 104 | time
+		ValidatorConsensusKeyRotationRecordQueue: collections.NewMap(
+			sb, types.ValidatorConsensusKeyRotationRecordIndexKey,
+			"cons_pub_rotation_queue",
```

Comment:
Keys are mixed up here, please double check:
```suggestion
		ValidatorConsensusKeyRotationRecordIndexKey: collections.NewMap(
			sb, types.ValidatorConsensusKeyRotationRecordIndexKey,
			"cons_pub_rotation_index",
			collections.PairKeyCodec(collections.BytesKey, sdk.TimeKey),
			collections.BytesValue,
		),

		// key format is: 104 | time
		ValidatorConsensusKeyRotationRecordQueue: collections.NewMap(
			sb, types.ValidatorConsensusKeyRotationRecordQueueKey,
			"cons_pub_rotation_queue",
```